RiotSwiftUI/Modules/Authentication/VerifyEmail/Coordinator/AuthenticationVerifyEmailCoordinator.swift

```diff
@@ -199,4 +212,109 @@ final class AuthenticationVerifyEmailCoordinator: Coordinator, Presentable {
         
         authenticationVerifyEmailViewModel.displayError(.unknown)
     }
+    
+    // Tchap: Add account creation part in this class
+    /// Creates an account on the homeserver with the supplied password.
+    @MainActor private func createAccount(password: String) {
+        let authService = AuthenticationService.shared
+        let registrationWizard = authService.registrationWizard
+        guard let registrationWizard = registrationWizard else {
+            MXLog.failure("[AuthenticationVerifyEmailCoordinator] createAccount: The registration wizard is nil.")
+            return
+        }
+        
+        startLoading()
+        
+        currentTask = Task { [weak self] in
+            do {
+                let result = try await registrationWizard.createAccount(username: nil,
+                                                                        password: password,
+                                                                        initialDeviceDisplayName: UIDevice.current.initialDisplayName)
+                
+                guard !Task.isCancelled else { return }
+                
+                switch result {
+                case .flowResponse(let flowResult):
+                    if flowResult.missingStages.contains(.email(isMandatory: true)) {
+                        authenticationVerifyEmailViewModel.context.send(viewAction: .send)
+                    } else {
+                        // Should not happen.
+                        MXLog.error("[AuthenticationVerifyEmailCoordinator] createAccount flowResponse with no e-mail missing stage !")
+                    }
+                case .success:
+                    MXLog.debug("[AuthenticationVerifyEmailCoordinator] createAccount success")
+                    self?.callback?(.completed(result))
+                }
+                
+                self?.stopLoading()
+            } catch {
+                self?.stopLoading()
+                self?.handleRegistrationError(error)
+            }
+        }
+    }
+    
+    /// Processes an error to either update the flow or display it to the user.
+    @MainActor private func handleRegistrationError(_ error: Error) {
+        if let mxError = MXError(nsError: error as NSError) {
+            authenticationVerifyEmailViewModel.displayError(.mxError(mxError.error))
```

Comment:
I think we will have to copy/paste here  a part of the conversion err_code -> error_message from https://github.com/tchapgouv/tchap-ios/pull/670/files#diff-8c29e966f3258d6c1a5cfb9e55799400c0584d7e733226ad7952a9db3502d3faL70

```
case kMXErrCodeStringPasswordTooShort:
                    message = TchapL10n.passwordPolicyTooShortPwdError
                case kMXErrCodeStringPasswordNoDigit:
                    message = TchapL10n.passwordPolicyWeakPwdError
                case kMXErrCodeStringPasswordNoLowercase:
                    message = TchapL10n.passwordPolicyWeakPwdError
                case kMXErrCodeStringPasswordNoUppercase:
                    message = TchapL10n.passwordPolicyWeakPwdError
                case kMXErrCodeStringPasswordNoSymbol:
                    message = TchapL10n.passwordPolicyWeakPwdError
                case kMXErrCodeStringWeakPassword:
                    message = TchapL10n.passwordPolicyWeakPwdError
                case kMXErrCodeStringPasswordInDictionary:
                    message = TchapL10n.passwordPolicyPwdInDictError
```

Reply:
Updated in last commits.